dropwizard-logging/src/test/java/io/dropwizard/logging/DefaultLoggingFactoryTest.java

```diff
@@ -71,6 +71,7 @@ public void canParseNewLoggerFormat() throws Exception {
         assertThat(fileAppenderFactory.getCurrentLogFilename()).isEqualTo("${new_app}.log");
         assertThat(fileAppenderFactory.getArchivedLogFilenamePattern()).isEqualTo("${new_app}-%d.log.gz");
         assertThat(fileAppenderFactory.getArchivedFileCount()).isEqualTo(5);
+        assertThat(fileAppenderFactory.getFilterFactories()).containsExactly(new TestFilterFactory(), new SecondTestFilterFactory());
```

Comment:
This currently produces a compiler warning about an unchecked call. What do you think about replacing this
check by a serie of checks?
```java
final ImmutableList<?> filterFactories = fileAppenderFactory.getFilterFactories();
assertThat(filterFactories).hasSize(2);
assertThat(filterFactories.get(0)).isExactlyInstanceOf(TestFilterFactory.class);
assertThat(filterFactories.get(1)).isExactlyInstanceOf(SecondTestFilterFactory.class);
```

As a benefit, we could get rid of `hashCode` and `equals` implementations for `TestFilterFactory` and `SecondTestFilterFactory`.